### ISSUE TITLE
Jest

### ISSUE THREAD

**Issue by venkatadry** [github/explore#1044]
Jest works well with other testing libraries like 
all

Jest is used to test _.
Java script

Jest is a JavaScript unit testing framework.
T

How to install Jest using npm?
npm install --save-dev jest

The below test will pass.
F

You can check strings against regular expressions with ______________ .
toMatch

Jest uses ________ and ________ matchers to test a value is with exact equality
ToBe , toEqual

The number can be compared using
all

To create a mock function, we use jest.fn().
T

______________ resets all information stored in the mockFn.mock.calls and mockFn.mock.instances arrays.
mockClear()

.mock is a property of mock function, the place where all information about how the function has been called is kept.
T

Jest supports ______________ for Isolating functionality into small, testable units under test.
Mock fn

The Asynchronous code will move on to another task before it finishes.
T

When you execute ______________ Code, it will move on to another task before it finishes.
asyn

Jest will wait until the ______________ callback is called before finishing the test.
done

______________ is useful when you want to mock functions in certain test cases and restore the original implementation in others.
mockRestore()

______________ resets all information stored in the mock, including any inital implementation given.
mockReset()

To test a particular function which throws an error when it's called, you can use toThrow.
T

Code coverage tests that how much code is covered under tests
T

______________ is useful when you want to completely restore a mock back to its initial state.
mockreset

The Synchronous Code will move on to another task before it finishes.
F
What is the command to install Jest and the babel-jest ?
$ npm install –dev-save jest babel-jest

$ npm install jest --save-dev babel-jestxxx

$ npm install --save-dev jest babel-jest

$ npm install -save-dev jest babel-jest

______________ returns the value only once when mock function is called.
mockReturnValue(value)xxx

mockReturnOnce(value)

mockReturnValueOnce(value)

None of the options

How to install Jest using yarn?
yarn add --dev jest

______________ is an array that records all the object instances that have been instantiated from the mock function using new.
mock.instances

Which method is useful to clean up a mock's usage data between two assertions.
mockClear()

mockRestore() removes the mock and restores the initial implementation
T

How to install Jest using npm?
npm install jest --save-dev

npm install --save-dev jest

npm add –save-dev jest

all the optionsxxxx
